survey_formio/models/survey.py

```diff
@@ -0,0 +1,149 @@
+
+    def _get_components(self, day_first=True):
+        """
+        Pages are not supported yet.
+
+        Returns a list of dict representing survey questions as form.io components
+
+        Parameters:
+        day_first: boolean that determines if the day comes first or
+                   not in dates
+        """
+        self.ensure_one()
+        components = []
+        for question in self.question_ids:
+            components.append(question._get_formio_dict(day_first))
+        return components
+
+    def generate_formio_json(self, day_first=True):
+        """
+        Returns a form.io conform json representation of a survey.
+
+        Parameters:
+        day_first: boolean that determines if the day comes first or
+                   not in dates
+        """
+        self.ensure_one()
+        display = self._get_formio_display()
+        title = self.title
+        components = self._get_components(day_first)
+
+        form_json = json.dumps(
+            {"display": display, "title": title, "components": components}
+        )
+        return form_json
+
+    def user_input_from_formio(self, formio_input, user_input_id=None, day_first=True):
+        """
+        If no user_input_id is given, creates a user_input with the values in formio_input.
+        If there is a user_input_id, updates that user_input with the values in formio_input.
+
+        Parameters:
+        formio_input: json given by form.io
+        user_input_id: survey.user_input id
+        day_first: boolean that determines if the day comes first or
+                   not in dates
+        """
+        formio_dict = json.loads(formio_input)
+        if not user_input_id:
+            user_input_id = (
+                self.env["survey.user_input"].create({"survey_id": self.id}).id
+            )
+        form_io_answers = formio_dict["data"]
+        for question in self.question_ids:
+            if form_io_answers.get(str(question.id)):
+                if question.question_type == "datetime":
+                    # Odoo expects a naive datetime but form.io gives a datetime with an offset.
+                    # We need to convert it to utc first, and then we replace the answer by
+                    # the utc datetime's string value
+
+                    # We can't use the fromisoformat function because it's only available
+                    # in Python v3.7+ and Odoo v13 is compatible with Python v3.6
+                    datetime_answer = form_io_answers[str(question.id)]
+                    form_io_answers[str(question.id)] = (
+                        datetime.strptime(datetime_answer[:19], "%Y-%m-%dT%H:%M:%S")
+                        + timedelta(
+                            hours=int(datetime_answer[20:22]),
+                            minutes=int(datetime_answer[23:]),
+                        )
+                        * (-1 if datetime_answer[19] == "+" else 1)
+                    ).strftime("%Y-%m-%d %H:%M:%S")
```

Comment:
Woud this work?

`datetime.datetime.strptime(datetime_answer, "%Y-%m-%dT%H:%M:%S%z").astimezone(pytz.UTC)`

Reply:
%z is not supported in Python 3.6